### PR TITLE
test: ethtypes: hex leading zeros

### PR DESCRIPTION
**Pull request by snissn**
## Related Issues
https://github.com/filecoin-project/lotus/issues/10170

## Proposed Changes
add tests that probe the ideas in the issue around Ethereum JSON-RPC responses should not include leading zero in hex
## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: chain/types/ethtypes/eth_types_test.go [hunk 1]
```diff
@@ -52,6 +55,7 @@ func TestEthBigIntMarshalJSON(t *testing.T) {
 		{EthBigInt(big.NewInt(65)), []byte("\"0x41\"")},
 		{EthBigInt(big.NewInt(1024)), []byte("\"0x400\"")},
 		{EthBigInt(big.Int{}), []byte("\"0x0\"")},
+		{EthBigInt(big.MustFromString("4314726344238778862224669783019953265742676409691641697694925658285796229120")), []byte("\"0x98a0c6ef2224342b2a8727ae5111fd000000000000000000000000000000000\"")},
```
Comment (magik6k) [inline chain/types/ethtypes/eth_types_test.go:58]: Should marshal without the leading zeroes

**Review by magik6k — COMMENTED** [review]
So as far as I understand, ethereum rpc expects `^0x([1-9a-f]+[0-9a-f]*|0)$`, so trimmed zeroes on the left side, and `0x0` for no value / all zero

This should apply to (almost?) all types in https://github.com/filecoin-project/lotus/blob/master/chain/types/ethtypes/eth_types.go which deal with the `0x` prefix.

The expected correct schema for each method/type is in https://github.com/filecoin-project/lotus/blob/master/itests/specs/eth_openrpc.json#L148

Essentially anything there that does something like
```go
"0x" + hex.EncodeToString(h[:])
```

should really do
```go
s :="0x" + strings.TrimLeft(hex.EncodeToString(h[:]), "0")
if s == "0x" { s = "0x0" }
```

File: chain/types/ethtypes/eth_types_test.go (hunk 1 above)
Comment (snissn) [reply chain/types/ethtypes/eth_types_test.go:58]: I can't construct an int that marshalls with leading zeros - are there any examples you can help me find?

**Comment by snissn** [comment]
moving to https://github.com/filecoin-project/lotus/pull/10261, still unclear next steps
